### ISSUE TITLE
Introduce a "stable" tag that points to the latest tagged Docker image version

### ISSUE THREAD

**Issue by whummer** [localstack/localstack#5340]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature description

We should consider introducing a `stable` tag that always points to the latest tagged version (e.g., `0.13.3`) of the image. Addresses the comment in this thread: https://github.com/localstack/localstack-java-utils/issues/80#issuecomment-1020028379

Also, we should discuss how to avoid mini-patches (x.x.x.x) from getting pushed to Docker Hub (mini-patches should only be a vehicle to publish updated packages to pypi). Either we only bump the `__version__` in `localstack/__init__.py` for at least patch releases (no x.x.x.x versions), or add a switch to the CI builds to skip pushing x.x.x.x tags. (addresses #5336)

/cc @thrau @alexrashed @dominikschubert

**Comment by thrau** [comment]
just very briefly my thoughts, we can drill into details as we go along:
* stable tag: in principle a good idea, although my opinion remains that it should be the other way around. `latest` should be stable and `nightly` should be the latest commit.
* avoid mini-patches getting pushed to docker hub: i'm assuming the reason for this issue is this recent commit: 7dab7b4db49943ad731796728d5f7516bc020653? it's regrettable that this broke #5336, apologies for that. i do think though that this is a symptom of the unclear/undocumented release process and the fact that, a) the pypi packages (CLI and shared localstack runtime code) and docker image share the same version number, while b) at the same time there's an invisible distinction between CLI, library code, and localstack docker image. i don't have a good suggestion for a better process, and i'm OK with this agreement for now.

  **_but_**: releasing something to pypi and not committing the version number increment into our `__init__.py` because that would trigger an undesirable re-push of the Dockerfile does not seem like the best way forward. that rather seems like a bug in the logic that pushes the tags, which seems easily fixable, right @alexrashed ?

* can we just drop the preceding zero already? ;-)

**Comment by calvernaz** [comment]
> latest -> updated for each commit on master (like today)

**Comment by alexrashed** [comment]
In principle I'm definitely in favor of introducing the stable tag.

**Concerning the naming:**
- I don't really have a strong opinion on `latest` vs `nightly` and there real convention in the Docker ecosystem (to my knowledge). However, in terms of migration, it might be interesting to acknowledge that a lot of users are already using the latest tag in their pipelines and environments:
  - If we want the majority of the user-base to switch to the release-based tag, we should use the `latest` tag for the last release and introduce a new `nightly` tag. 
  - If we don't want to break the semantics, we should introduce a `stable` tag and keep the `latest` tag as it is.

**Concerning the mini-patches:**
The already released docker tag has been updated due to the fact that a sub-version was released. So version 0.13.3 has been updated because version 0.13.3.x has been released.
Even though we do not commit to semantic-versioning officially, if we would compare our versioning with semantic-versioning one could maybe drop the initial 0. This would mean we update the tag to a minor version ("13.3") because a patch version ("13.3.1") has been released (which would perfectly fine, even in semantic versioning). We also do that for our "major" versions (i.e. we update the `0.13` tag when `0.13.x` is released).
So, I agree with @thrau that it's a symptom of a missing distinction and unclear release processes (especially the non-explicitly declared "mini-patch" versions).
But I think it could also be dangerous to not update a version tag if we release a "mini-patch".
And it would definitely re-introduce https://github.com/localstack/localstack/issues/5032.
It would, however, not be too hard to change the `Makefile` such that it does not update the tag in case of "mini-patches".

**Concerning dropping the 0:**
It would make things a lot more clearer (What is a mini-patch?), and I would also be in favor of going that path.
However, versions like "1.2.3" are more likely to be associated with semantic versioning.
Even though we do _not_ state that we comply to semantic versioning, a lot of users will assume that implicitly.

**TL/DR:**
- Add a `stable` or `nightly` tag depending on a preferred user migration path.
- Not updating Docker images when releases are published could be dangerous and would re-introduce https://github.com/localstack/localstack/issues/5032.
- I would be in favor of making the release process as clearly defined as possible, which would make it obvious what I get / pin / download (for developers and users).
- Since changing our release process shouldn't be done over night, we _could_ - until then - change the Makefile such that we don't publish mini-patches (on the cost of https://github.com/localstack/localstack/issues/5032).
